x-pack/legacy/plugins/siem/public/pages/case/components/configure_cases/index.test.tsx

```diff
@@ -0,0 +1,756 @@
+    expect(newWrapper.find(Mapping).prop('disabled')).toBe(true);
+    expect(newWrapper.find(Mapping).prop('updateConnectorDisabled')).toBe(true);
+  });
+
+  test('it shows the warning callout when configuration is invalid', () => {
+    useCaseConfigureMock.mockImplementation(
+      ({ setConnector, setClosureType, setCurrentConfiguration }) => {
+        useEffect(() => setConnector('not-id'), []);
+        return useCaseConfigureResponse;
+      }
+    );
+
+    const newWrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+
+    expect(
+      newWrapper.find('[data-test-subj="configure-cases-warning-callout"]').exists()
+    ).toBeTruthy();
+  });
+
+  test('it disables correctly Connector when loading connectors', () => {
+    useConnectorsMock.mockImplementation(() => ({
+      ...useConnectorsResponse,
+      loading: true,
+    }));
+
+    const newWrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+
+    expect(newWrapper.find(Connectors).prop('disabled')).toBe(true);
+  });
+
+  test('it disables correctly Connector when saving configuration', () => {
+    useCaseConfigureMock.mockImplementation(() => ({
+      ...useCaseConfigureResponse,
+      persistLoading: true,
+    }));
+
+    const newWrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+
+    expect(newWrapper.find(Connectors).prop('disabled')).toBe(true);
+  });
+
+  test('it pass the correct value to isLoading attribute on Connector', () => {
+    useConnectorsMock.mockImplementation(() => ({
+      ...useConnectorsResponse,
+      loading: true,
+    }));
+
+    const newWrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+
+    expect(newWrapper.find(Connectors).prop('isLoading')).toBe(true);
+  });
+
+  test('it set correctly the selected connector', () => {
+    useCaseConfigureMock.mockImplementation(
+      ({ setConnector, setClosureType, setCurrentConfiguration }) => {
+        useEffect(() => setConnector('456'), []);
+        return useCaseConfigureResponse;
+      }
+    );
+
+    const newWrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+
+    expect(newWrapper.find(Connectors).prop('selectedConnector')).toBe('456');
+  });
+
+  test('the connector is changed successfully', () => {
+    wrapper.find('button[data-test-subj="dropdown-connectors"]').simulate('click');
+    wrapper.update();
+    wrapper.find('button[data-test-subj="dropdown-connector-456"]').simulate('click');
+    wrapper.update();
+
+    expect(wrapper.find(Connectors).prop('selectedConnector')).toBe('456');
+  });
+
+  test('the connector is changed successfully to none', () => {
+    wrapper = mount(<ConfigureCases userCanCrud />, { wrappingComponent: TestProviders });
+    wrapper.find('button[data-test-subj="dropdown-connectors"]').simulate('click');
```

Comment:
If it is possible, I would test the detail behaviour of a component in its own unit test, and skip it here.